src/udev/udev-builtin-net_id.c

```diff
@@ -349,6 +352,50 @@ static int names_vio(struct udev_device *dev, struct netnames *names) {
         return 0;
 }
 
+#define _PLATFORM_TEST "/sys/devices/platform/vvvvPPPP"
+#define _PLATFORM_PATTERN4 "/sys/devices/platform/%4s%4x:%2x/net/eth%u"
+#define _PLATFORM_PATTERN3 "/sys/devices/platform/%3s%4x:%2x/net/eth%u"
+
+static int names_platform(struct udev_device *dev, struct netnames *names, bool test) {
+        struct udev_device *parent;
+        char vendor[5];
+        char shortid;
+        unsigned model, instance, ethid;
+        const char *syspath, *pattern;
+
+        /* check if our direct parent is a platform device with no other bus in-between */
+        parent = udev_device_get_parent(dev);
+        if (!parent)
+                return -ENOENT;
+
+        if (!streq_ptr("platform", udev_device_get_subsystem(parent)))
+                 return -ENOENT;
+
+        syspath = udev_device_get_syspath(dev);
+
+        /* Vendor ID inside ACPI id can be either 3 or 4 */
+        if (syspath[sizeof _PLATFORM_TEST - 1] == ':')
+                pattern = _PLATFORM_PATTERN4;
+        else
+                pattern = _PLATFORM_PATTERN3;
+
+        /* Platform devices are named after ACPI table match, and instance id
+         * eg. "/sys/devices/platform/HISI00C2:00");
+         * The Vendor (3 or 4 char), followed by hexdecimal model number : instance id.
+         */
+        if (sscanf(syspath, pattern, vendor, &model, &instance, &ethid) != 4)
+                 return -EINVAL;
+
+        if (streq(vendor, "HISI"))
+                shortid = 'h';
+        else
+                return -ENOENT;
```

Comment:
why check for "HISI"? wouldn't this scheme work for all kinds of ACPI networking devices automatically? Why limit it like this?

If it does make sense to check exclusively for HISI, then there should be a brief explanation what that actually means, i.e. which kind of devices are precisely matched here, and why this doesn't apply to anything else.